control.py

```diff
@@ -0,0 +1,147 @@
+"""
+Contains useful commands to work with the bot and the server. Used by make tool.
+It relies on some linux-specific logic, and thus might not work in another OS.
+"""
+
+import sys
+import time
+import argparse
+from server.read_config import config
+from ai_replica.engine.reconstruct_mind import reconstruct
+from ai_replica.utils.system import (
+  execute_command,
+  open_url_in_browser,
+  is_expected_python_version,
+  get_python_exec_name,
+  install_replica_dependencies,
+)
+from rasa.utils import (
+  start_rasa_actions_server,
+  start_rasa_main_server,
+  train_model,
+  stop_a_rasa_server,
+  install_rasa_dependencies,
+  get_paths as get_rasa_paths,
+)
+from server_bot import stop_server as stop_basic_server
+
+
+rasa_required_python = config["system"]["rasa_preferred_python_version"]
+expected_launch_duration_sec = config["rasa"]["expected_launch_duration_sec"]
+
+
+def print_notifications():
+  if "linux" in sys.platform and is_expected_python_version(rasa_required_python):
+    print("Detected the right OS and python version, which is nice.")
+  else:
+    print(
+      f"Note: this script was tested in linux and Python {rasa_required_python}"
+    )
+    print("It might work incorrectly in other environments.")
+    print("Note: if you close this terminal, the server stuff will continue to run")
+
+
+def start_rasa_server(rasa_path, work_dir):
+  stop_a_rasa_server(rasa_path, server_name="main")
+  start_rasa_main_server(rasa_path, work_dir)
+
+
+def start_rasa_server_with_debug_logs(rasa_path, work_dir):
+  stop_a_rasa_server(rasa_path, work_dir, server_name="main")
+  start_rasa_main_server(
+    rasa_path,
+    work_dir,
+    "--debug",
+  )
+
+
+def restart_servers(rasa_path, work_dir, launch_bools):
+  if any(launch_bools.values()):
+    print(f"Restarting only the specified servers: {launch_bools}")
+  else:
+    print("You didn't specify which server to restart, so I restart them all.")
+    print("I'll also retrain the model.")
+    launch_bools = dict.fromkeys(launch_bools, True)
+    train_model(rasa_path, work_dir)
+
+    if launch_bools["main_rasa"]:
+      stop_a_rasa_server(rasa_path, server_name="main")
+      start_rasa_main_server(rasa_path, work_dir)
+
+    if launch_bools["actions_rasa"]:
+      stop_a_rasa_server(rasa_path, server_name="actions")
+      start_rasa_actions_server(rasa_path, work_dir)
+
+    if launch_bools["basic"]:
+      stop_basic_server()
+      build_web_chat()
+      python_name = get_python_exec_name()
+      execute_command(python_name, "server_bot.py", run_in_another_terminal7=True)
```

Comment:
The lines 67-79 should be moved one level to the left. 

The code in this func works as follows:

- the first else/if (lines 59-65) decides what to do with `launch_bools`
- the rest (lines 67-79) is executing `launch_bools`

Reply:
Fixed.